common/lcs/src/ser.rs

```diff
@@ -132,6 +136,14 @@ where
         }
         self.output_u32_as_uleb128(len as u32)
     }
+
+    fn enter_named_container(&mut self, name: &'static str) -> Result<()> {
+        if self.max_remaining_depth == 0 {
+            return Err(Error::ExceededObjectDepthLimit(name));
+        }
+        self.max_remaining_depth -= 1;
```

Comment:
Could one initialize `max_remaining_depth` with a large negative number, and make this underflow to stg large on first iteration? Do you want to use `checked_sub`?

Reply:
`max_remaining_depth ` is a `usize` so there are no negative values.

`checked_sub` is a good idea but it returns an option so I was not sure about if the generated assembly was as good. (Let me check)

Reply:
You're correct ... in debug mode. 
https://play.rust-lang.org/?version=stable&mode=release&edition=2018&gist=4543f51666cebea902bd639690440012

Reply:
:facepalm: Indeed, you're playing within ℤ/18446744073709551615ℤ and with decrements of size 1 you could only wrap to a larger number from 0, which is explicitly excluded, my bad.

Reply:
:facepalm: Indeed, you're playing within ℤ/(18446744073709551615ℤ) and with decrements of size 1 you could only wrap to a larger number from 0, which is explicitly excluded, my bad.